app/uk/gov/hmrc/exports/services/reversemapping/declaration/ExportsDeclarationXmlParser.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.services.reversemapping.declaration
+
+import uk.gov.hmrc.exports.models.DeclarationType
+import uk.gov.hmrc.exports.models.declaration._
+import uk.gov.hmrc.exports.services.reversemapping.declaration.items.ItemsParser
+
+import java.time.Instant
+import java.util.UUID
+import javax.inject.Inject
+import scala.xml.NodeSeq
+
+class ExportsDeclarationXmlParser @Inject()(
+  additionalDeclarationTypeParser: AdditionalDeclarationTypeParser,
+  consignmentReferencesParser: ConsignmentReferencesParser,
+  mucrParser: MucrParser,
+  itemsParser: ItemsParser
+) {
+
+  def fromXml(xml: String): ExportsDeclaration = {
+    val declarationXml = scala.xml.XML.loadString(xml)
+
+    buildExportsDeclaration(declarationXml)
+  }
+
+  private def buildExportsDeclaration(declarationXml: NodeSeq): ExportsDeclaration =
+    ExportsDeclaration(
+      id = UUID.randomUUID().toString,
+      eori = "GB1234567890",
```

Comment:
Just to be sure... is this the assignment we really want here?

Reply:
There needs to be a placeholder for now. Rules around this eori are not obvious. I am not even sure, if we will not need to take it from authentication instead.